File: WebApplication/src/modules/operator/dtos/operator-search-result.dto.ts
import { OperatorType } from "../../application/application-one/enums/operator-type.enum";
import { Authority } from "../../nomenclature/models/authority.model";
import { Settlement } from "../../nomenclature/models/settlement/settlement.model";

export class OperatorSearchResultDto {
  id: number;

  type: OperatorType;

  fullname: string;
  legalEntityUic: string;

  authorityRiosvId: number;
  authorityRiosv: Authority;

  authorityBasinId: number;
  authorityBasin: Authority;

  settlementId: number;
  settlement: Settlement;
}

File: Zopoesht.Application/Operators/Dtos/OperatorSearchResultDto.cs
using Zopoesht.Data.Nomenclatures;
using Zopoesht.Data.Nomenclatures.Operators.Enums;
using Zopoesht.Data.Nomenclatures.Settlements;

namespace Zopoesht.Application.Operators.Dtos
{
    public class OperatorSearchResultDto
    {
        public int Id { get; set; }
        public string Fullname { get; set; }
        public OperatorType? Type { get; set; }
        public string LegalEntityUic { get; set; }
        public int? SettlementId { get; set; }
        public Settlement Settlement { get; set; }
        public int? AuthorityRiosvId { get; set; }
        public Authority AuthoriyRiosv { get; set; }
        public int? AuthorityBasinId { get; set; }
        public Authority AuthorityBasin { get; set; }
    }
}
